### PR TITLE
Applied Spotless and Detekt to the "feature-datatable" module. MIFOSAC-282

### PR DESCRIPTION
**Pull request by Aditya3815**
Apply Spotless and Detekt to the "feature-datatable" module.

Fixes https://mifosforge.jira.com/browse/MIFOSAC-282

Please Add Screenshots If there are any UI change
Screenshot 2024-12-07 225832
Screenshot 2024-12-07 225909

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by Aditya3815** [comment]
Sir, please merge this PR this is done

**Comment by itsPronay** [comment]
@Aditya3815 please fix merge conflicts

File: feature/data-table/src/main/java/com/mifos/feature/dataTable/dataTable/DataTableScreen.kt [hunk 1]
```diff
@@ -88,19 +98,19 @@ fun DataTableScreen(
         icon = MifosIcons.arrowBack,
         title = stringResource(id = R.string.feature_data_table_title),
         onBackPressed = navigateBack,
-        snackbarHostState = snackbarHostState
+        snackbarHostState = snackbarHostState,
     ) {
         Box(
-            modifier = Modifier
+            modifier = modifier // Pass the modifier here
```
Comment (itsPronay) [inline feature/data-table/src/main/java/com/mifos/feature/dataTable/dataTable/DataTableScreen.kt:104]: can you remove comments here?

File: feature/data-table/src/main/java/com/mifos/feature/dataTable/dataTableData/DataTableDataScreen.kt [hunk 2]
```diff
@@ -162,17 +168,16 @@ fun DataTableDataScreen(
                 Icon(imageVector = MifosIcons.Add, contentDescription = null)
             }
         },
-        snackbarHostState = snackbarHostState
+        snackbarHostState = snackbarHostState,
     ) { paddingValues ->
-        Column(modifier = Modifier.padding(paddingValues)) {
+        Column(modifier = modifier.padding(paddingValues)) { // Use the passed Modifier here
```
Comment (itsPronay) [inline feature/data-table/src/main/java/com/mifos/feature/dataTable/dataTableData/DataTableDataScreen.kt:173]: here as well

**Review by itsPronay — CHANGES_REQUESTED** [review]

**Comment by Aditya3815** [comment]
Sure

**Pull request merged**